core/src/test/java/org/frankframework/pipes/IfPipeJsonPathTest.java

```diff
@@ -0,0 +1,110 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.util.CloseUtils;
+
+public class IfPipeJsonPathTest extends PipeTestBase<IfPipe> {
+
+	private PipeRunResult pipeRunResult;
+
+	@Override
+	@AfterEach
+	public void tearDown() {
+		CloseUtils.closeSilently(pipeRunResult);
+
+		super.tearDown();
+	}
+
+	@Override
+	public IfPipe createPipe() throws ConfigurationException {
+		IfPipe ifPipe = new IfPipe();
+
+		// Add default forwards
+		ifPipe.addForward(new PipeForward(PIPE_FORWARD_THEN, null));
+		ifPipe.addForward(new PipeForward(PIPE_FORWARD_ELSE, null));
+
+		return ifPipe;
+	}
+
+	static String testJson = """
+			{
+			    "store": {
+			        "book": [
+			            {
+			                "category": "reference",
+			                "author": "Nigel Rees",
+			                "title": "Sayings of the Century",
+			                "price": 8.95
+			            },
+			            {
+			                "category": "fiction",
+			                "author": "Evelyn Waugh",
+			                "title": "Sword of Honour",
+			                "price": 12.99
+			            },
+			            {
+			                "category": "fiction",
+			                "author": "Herman Melville",
+			                "title": "Moby Dick",
+			                "isbn": "0-553-21311-3",
+			                "price": 8.99
+			            },
+			            {
+			                "category": "fiction",
+			                "author": "J. R. R. Tolkien",
+			                "title": "The Lord of the Rings",
+			                "isbn": "0-395-19395-8",
+			                "price": 22.99
+			            }
+			        ],
+			        "bicycle": {
+			            "color": "red",
+			            "price": 19.95
+			        }
+			    },
+			    "expensive": 10
+			}
+			""";
+
+	public static Stream<Arguments> messageSource() {
+		return Stream.of(
+				// input, expression, expressionValue, expectedValue
+				Arguments.of("{root: ''}", "$.root", "", PIPE_FORWARD_THEN),
+				Arguments.of("{root: 'test'}", "$.root", "", PIPE_FORWARD_THEN),
+				Arguments.of("{root: ''}", "$.root", "test", PIPE_FORWARD_ELSE),
+				Arguments.of("{root: 'test'}", "$.root", "test", PIPE_FORWARD_THEN),
+				Arguments.of("{root: 'test123'}", "$.root", "test", PIPE_FORWARD_ELSE),
+				Arguments.of(testJson, "$.store.book[1].author", "", PIPE_FORWARD_THEN),
+				Arguments.of(testJson, "$.store.book[1].author", "Evelyn Waugh", PIPE_FORWARD_THEN),
+				Arguments.of(testJson, "$.store.book[1].isbn", "", PIPE_FORWARD_ELSE),
+				Arguments.of(testJson, "$.store.book[1].category", "reference", PIPE_FORWARD_ELSE)
+		);
+
+		// TODO check if jsonPath is actually used expression based. Like if "priceOfAllBooks > x" evaluates to true? I'm not sure if jsonpath can do that
```

Comment:
Open end: it seems that expressions like `number(count(/results/result[contains(@name , 'test')]))>1` as used in Xpath/XSL aren't possible in JsonPath. Is this something we really need here for the jsonPath support?

Reply:
That works for us because we don't use an actual xPath but an xslt.